VideoLocker/src/androidTest/java/org/edx/mobile/test/http/MockApiTests.java

```diff
@@ -0,0 +1,209 @@
+        Config.setInstance(new Config(properties));
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        server.shutdown();
+    }
+
+    /**
+     * Randomly determine whether this call should result in a network failure.
+     */
+    private static boolean calculateIsFailure() {
+        int randomValue = random.nextInt(100) + 1;
+        return randomValue <= ERROR_PCT;
+    }
+
+    // The delay randomizing methods below are copied from the Retrofit
+    // MockRestAdapter implementation which is distributed under the Apache 2.0
+    // License
+    /**
+     * Get the delay (in milliseconds) that should be used for triggering a
+     * network error.
+     * <p>
+     * Because we are triggering an error, use a random delay between 0 and
+     * three times the normal network delay to simulate a flaky connection
+     * failing anywhere from quickly to slowly.
+     */
+    private static int calculateDelayForError() {
+        return random.nextInt(DELAY_MS * ERROR_DELAY_FACTOR);
+    }
+
+    /**
+     * Get the delay (in milliseconds) that should be used for delaying
+     * a successful network call response.
+     */
+    private static int calculateDelayForSuccess() {
+        float errorPercent = VARIANCE_PCT / 100f; // e.g., 20 / 100f == 0.2f
+        float lowerBound = 1f - errorPercent; // 0.2f --> 0.8f
+        float upperBound = 1f + errorPercent; // 0.2f --> 1.2f
+        float bound = upperBound - lowerBound; // 1.2f - 0.8f == 0.4f
+        float delayPercent = (random.nextFloat() * bound) +
+                lowerBound; // 0.8 + (rnd * 0.4)
+        return (int) (DELAY_MS * delayPercent);
+    }
+
+    /**
+     * Get the delay (in milliseconds) that should be used for delaying
+     * a network call response.
+     */
+    private static int calculateDelayForCall() {
+        // Commenting out the random failure mode delay since we want our
+        // tests to be reproducible
+        return //calculateIsFailure() ? calculateDelayForError() :
+                calculateDelayForSuccess();
+    }
+
+    /**
+     * Match url to a regex template while allowing extra slash and query
+     * strings at the end
+     */
+    private static boolean urlMatches(String url, String template) {
+        if (TextUtils.isEmpty(url) || TextUtils.isEmpty(template)) {
+            return false;
+        }
+        String pattern = '^' + template;
+        if (template.charAt(template.length() - 1) != '/') {
+            pattern += "/?";
+        }
+        pattern += "(\\?.*)?$";
+        return url.matches(pattern);
+    }
+
+    /**
+     * Handler for requests on the mock server that will send mock responses
+     * according to the request urls
+     */
+    private class MockResponseDispatcher extends Dispatcher {
+        @Override
+        public MockResponse dispatch(RecordedRequest recordedRequest)
+                throws InterruptedException {
```

Comment:
Sure we can, but it would need similar input (url, method, body, etc.). So what would be the point in delegating to an external method/class?